mesonpy/__init__.py

```diff
@@ -1227,7 +1227,7 @@ def get_requires_for_build_wheel(
 ) -> List[str]:
     dependencies = []
 
-    if _env_ninja_command() is None:
+    if os.environ.get('NINJA') is None and _env_ninja_command() is None:
```

Comment:
Changed per #316

I was trying to figure out why this wasn't an issue if we just don't add it as a build-requires. The answer is it depends on the previous commit raising a `ConfigError()` that is logically part of *this* commit.

Reply:
I'm not sure I follow. Taking this commit in isolation, before this commit, if the `$NINJA` environment variable pointed to an invalid ninja executable, we would still require the `ninja` package but we would have left the environment variable pointing to the wrong executable and meson would have reported that it cannot find ninja. With this commit nothing changes in this respect, we just do not request to install a ninja package that would not be used anyway (because meson obeys the `$NINJA` environment variable).